packages/react-native/ReactCommon/react/nativemodule/idlecallbacks/IdleTaskRef.cpp

```diff
@@ -0,0 +1,17 @@
+/*
+* Copyright (c) Meta Platforms, Inc. and affiliates.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+#include "IdleTaskRef.h"
+
+namespace facebook::react {
+
+IdleTaskRef::IdleTaskRef(std::shared_ptr<Task> task): task(task) {}
```

Comment:
Nit: I just thought that this class could be defined inline in `NativeIdleCallbacks.cpp` because it's only used there (and it'd be good if it was just an implementation detail for that class).